internal/drain-config/drainConfig.go

```diff
@@ -189,9 +188,6 @@ func setupDBConn(conf *config.Config) (*sql.DB, error) {
 	if err != nil {
 		return nil, fmt.Errorf("db open: %v", err)
 	}
-	if err := db.Ping(); err != nil {
-		return nil, fmt.Errorf("db ping: %v", err)
-	}
```

Comment:
Don't we need this?

Reply:
No we will just setup the connection pool and not ping the db while setting up. Migrations will fail if DB connection fails.

Reply:
Don't we need this for Gateway and Processor?

Reply:
For processor also, migrations will fail if DB connection fails. Processor will handle the error and setup will fail, gateway will just log the error and continue to ingest events.